brokers/layouts/layouts.go

```diff
@@ -0,0 +1,239 @@
+package layouts
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ubuntu/authd/internal/testsdetection"
+)
+
+// UIType is the type to define an authd UI layout type.
+type UIType int
+
+const (
+	// UIForm is the type of a Form UI layout type.
+	UIForm UIType = iota
+	// UIQrCode is the type of a QrCode UI layout type.
+	UIQrCode
+	// UINewPassword is the type of a NewPassword UI layout type.
+	UINewPassword
+
+	// These are only for testing purposes, so we don't expose them publicly.
+	uiRequiredEntry UIType = iota + 1000
+	uiOptionalEntry
+)
+
+// UITypeError defines an error for [UIType] errors.
+type UITypeError struct {
+	error
+}
+
+// Is makes this error insensitive to the actual error content.
+func (UITypeError) Is(err error) bool { return err == UITypeError{} }
+
+func (u UIType) String() string {
+	switch u {
+	case UIForm:
+		return Form
+	case UIQrCode:
+		return QrCode
```

Comment:
I didn't rename these on purpose (I didn't want to change behavior), but I think it's now a good moment for doing it... So likely this should be now called externally `WebAuth` or something similar (I wouldn't call it _device_ auth since that's not dependent on any device per se, but rather on a web authentication triggered through the content URI).